flask-server/main.py

```diff
@@ -425,6 +425,78 @@ def fetch_username(user_id):
     key = datastore_client.key('user', int(user_id))
     user = list(query.add_filter('__key__', '=', key).fetch(limit=1))[0]
     return user['username'], 200
+    
+# edit bookmark api
+@app.route('/apis/edit-bookmark', methods=['POST'])
+def edit_bookmark():
+    resp_token = decode_auth_token(request.json['Authorization'])
+
+    if is_valid_instance(resp_token):
+        link = request.json['link']
+        title = request.json['title']
+        description = request.json['description']
+
+        key = datastore_client.key('bookmark', int(request.json['bookmarkId']))
+        bookmark = list(datastore_client.query(kind='bookmark').add_filter('__key__', '=', key).add_filter('deleted', '=', False).fetch(limit=1))[0]
+        
+        if link:
+            bookmark.update({
+                'link': link
+            })
+        if title:
+            bookmark.update({
+                'title': title
+            })
+        if description: 
+            bookmark.update({
+                'description': description
+            })
+
+        datastore_client.put(bookmark)
+
+        response_object = {
+            'link': bookmark['link'],
+            'title': bookmark['title'],
+            'description': bookmark['description'],
+            'id': bookmark.id
+        }
+        return response_object, 200
+    else: 
+        response_object = {
+            'status': 'fail',
+            'message': 'Invalid JWT. Failed to edit bookmark.'
+        }
+        return response_object, 401
+
+# edit clink api
+@app.route('/apis/edit-clink', methods=['POST'])
+def edit_clink():
+    resp_token = decode_auth_token(request.json['Authorization'])
+
+    if is_valid_instance(resp_token):
+        title = request.json['title']
+
+        key = datastore_client.key('clink', int(request.json['clinkId']))
+        clink = list(datastore_client.query(kind='clink').add_filter('__key__', '=', key).add_filter('deleted', '=', False).fetch(limit=1))[0]
```

Comment:
does datastore_client.get(key) not work?

Reply:
why would i use that?

Reply:
https://cloud.google.com/datastore/docs/concepts/entities#updating_an_entity

Reply:
bet